### PR TITLE
feat(cli): add `bindle clean` subcommand to clear the local cache

### LINKED ISSUE #280: Add a `bindle clean` subcommand
Right now the bindle cli uses the `DumbCache` to store files locally. There are many reasons to want to clear this cache, particularly if you are testing bindle locally and nuke your bindle server's store of data. 

We should add a `clean` subcommand to the bindle CLI that removes all cached bindles

### PR DESCRIPTION
**Pull request by VishnuJin**
fixes #280 
This PR adds `clean` subcommand to `bindle cli` to clear the DumpCache

### CODE REVIEW

File: bin/client/main.rs [hunk 1]
```diff
@@ -433,6 +433,10 @@ async fn run() -> std::result::Result<(), ClientError> {
                 }
             }
         }
+        SubCommand::Clean(_clean_opts) => {
+            // TODO: Clean up cached bindles
```
Comment (thomastaylor312) [inline bin/client/main.rs:437]: You missing a commit here?

Comment (thomastaylor312) [reply bin/client/main.rs:437]: Oh, just saw you move it to draft. Never mind

Comment (VishnuJin) [reply bin/client/main.rs:437]: @thomastaylor312 sorry Im working on it, this is still in draft

**Comment by VishnuJin** [comment]
hi @thomastaylor312 some brief hints to proceed further on this implementation would be helpful !!

- should `clean` have any args with it
- should we delete everything in the cache dir

**Comment by thomastaylor312** [comment]
I think for this first implementation we don't need any extra flags (other than the top level `bindle-dir` which is included by default). Deleting all the things in that cache dir is also just fine for now!

File: bin/client/main.rs [hunk 2]
```diff
@@ -433,6 +433,19 @@ async fn run() -> std::result::Result<(), ClientError> {
                 }
             }
         }
+        SubCommand::Clean(_clean_opts) => {
+            // Cleans up the local bindles directory.
+            println!("Cleaning up bindles in {:?}...", bindle_dir);
```
Comment (thomastaylor312) [inline bin/client/main.rs:438]: Nit: It is generally better to just use `{}` and then use the display implementation of `PathBuf`: `bindle_dir.display()`

File: bin/client/main.rs [hunk 3]
```diff
@@ -145,7 +145,7 @@ async fn run() -> std::result::Result<(), ClientError> {
         .build(&opts.server_url, token, keyring)?;
 
     let local = bindle::provider::file::FileProvider::new(
-        bindle_dir,
+        bindle_dir.clone(),
```
Comment (thomastaylor312) [inline bin/client/main.rs:148]: Nit: I think you can just pass a borrow here actually

**Review by thomastaylor312 — APPROVED** [review]
Looking good! Just a few optional nits! I'll give it a day in case you'd like to address them and then I'll merge!

File: bin/client/main.rs (hunk 3 above)
Comment (VishnuJin) [reply bin/client/main.rs]: fixed

File: bin/client/main.rs (hunk 2 above)
Comment (VishnuJin) [reply bin/client/main.rs]: fixed

**Pull request merged**